Source/Core/VideoBackends/Vulkan/CommandBufferManager.cpp

```diff
@@ -308,6 +308,15 @@ void CommandBufferManager::SubmitCommandBuffer(bool submit_on_worker_thread,
     }
   }
 
+  // When we call BeginCommandBuffer() later, we'll attempt to wait for the completion of the next
+  // command buffer.  But WaitForCommandBufferCompletion requires the submission thread to be idle
+  // before it does anything, which means it'll also end up waiting for it to finish processing
+  // *this* command buffer.  Which kind of prevents backend multithreading from actually
+  // multithreading.  Work around that by doing the wait *before* we submit this command buffer.
+  const u32 next_buffer_index = (m_current_cmd_buffer + 1) % NUM_COMMAND_BUFFERS;
```

Comment:
Can we use `next_buffer_index` and remove `cmd_buffer_index`?  Doesn't look like `m_current_cmd_buffer` changes.

Reply:
We don't advance `m_current_cmd_buffer` here, that happens in BeginCommandBuffer

Either way, I just realized it shouldn't be that hard to prevent from waiting for the submit thread, going to redo this